package/MDAnalysis/lib/pkdtree.py

```diff
@@ -28,199 +28,204 @@
+          `leafsize` will speed up the construction of the KDTree but
           slow down the search.
-        """
-        self.dim = 3  # 3D systems
-        self.box = None
-        self._dm = None  # matrix of central-cell vectors
-        self._rm = None  # matrix of normalized reciprocal vectors
-        self.initialize_bm(box)
-        self.kdt = _CKDTree.KDTree(self.dim, bucket_size)
-        self._indices = list()
-
-    def initialize_bm(self, box):
-        """
-        Store box information and define direct and reciprocal box matrices.
-        Rows of the direct matrix are the components of the central cell
-        vectors. Rows of the reciprocal matrix are the components of the
-        normalized reciprocal lattice vectors. Each row of the reciprocal
-        matrix represents the vector normal to the unit cell face
-        associated to each axis.
-        For instance, in an orthorhombic cell the YZ-plane is associated to
-        the X-axis and its normal vector is (1, 0, 0). In a triclinic cell,
-        the plane associated to vector ``\vec{a}`` is perpendicular to the
-        normalized cross product of ``\vec{b}`` and ``\vec{c}``.
 
-        Parameters
-        ----------
-        box : array-like or ``None``, optional, default ``None``
-          Simulation cell dimensions in the form of
-          :attr:`MDAnalysis.trajectory.base.Timestep.dimensions` when
-          periodic boundary conditions should be taken into account for
-          the calculation of contacts.
         """
-        box_type = _box_check(box)
-        if box_type == 'ortho':
-            a, b, c = box[:3]
-            dm = np.array([[a, 0, 0], [0, b, 0], [0, 0, c]], dtype=np.float32)
-            rm = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]], dtype=np.float32)
-        elif box_type in 'tri_box tri_vecs tri_vecs_bad':
-            if box_type == 'tri_box':
-                dm = triclinic_vectors(box)
-            elif box_type == 'tri_vecs':
-                dm = box.copy('C')
-            else:  # case 'tri_vecs_bad'
-                dm = triclinic_vectors(triclinic_box(box[0], box[1], box[2]))
-            rm = np.zeros(9, dtype=np.float32).reshape(3, 3)
-            rm[0] = np.cross(dm[1], dm[2])
-            rm[1] = np.cross(dm[2], dm[0])
-            rm[2] = np.cross(dm[0], dm[1])
-            for i in range(self.dim):
-                rm[i] /= norm(rm[i])  # normalize
-        else:
-            raise ValueError('Failed to initialize direct/reciprocal matrices')
+        self.leafsize = leafsize
+        self.dim = 3  # 3D systems
         self.box = box
-        self._dm = dm
-        self._rm = rm
+        self._pbc = False
+        if self.box is not None:
+            self._pbc = True
+        self._built = False
+        self.cutoff = None
 
-    def set_coords(self, coords):
-        """
-        Add coordinates of the points. Wrapping of coordinates to the central
-        cell is enforced along the periodic axes.
+    @property
```

Comment:
So what you can do with a property is make it dynamic, so:
```python
@property
def pbc(self):
    return self.box is not None
```

This makes it act like a property, but the information can never fall out of date because it's generated on demand.  This helps keep the code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
Don't worry about setters/getters for this